### ISSUE TITLE
`@everywhere using` loads module twice on the workers

### ISSUE THREAD

**Issue by amitmurthy** [JuliaLang/julia#19187]
M.jl
```
module M
  println("Loaded M on $(myid())")
end
```

julia -p 4
```

julia> push!(LOAD_PATH, ".")

julia> @everywhere using M
Loaded M on 1
	From worker 4:	Loaded M on 4
	From worker 3:	Loaded M on 3
	From worker 2:	Loaded M on 2
	From worker 5:	Loaded M on 5
WARNING: replacing module M
WARNING: replacing module M
	From worker 5:	Loaded M on 5
	From worker 2:	Loaded M on 2
WARNING: replacing module M
	From worker 4:	Loaded M on 4
WARNING: replacing module M
	From worker 3:	Loaded M on 3
```

Needs to be fixed on 0.5

On 0.6 https://github.com/JuliaLang/julia/pull/19073 is pending review.

**Comment by amitmurthy** [comment]
Closed by #21718

**Comment by gaika** [comment]
```
[joe@z1 julia]$ ./julia -p 3
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.7.0-DEV.4881 (2018-04-13 22:09 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit b6d81e3 (0 days old master)
|__/                   |  x86_64-redhat-linux

julia> using Distributed

julia> @everywhere using Colors
┌ Warning: Replacing module `Colors`
└ @ Base loading.jl:894
┌ Warning: Replacing module `Colors`
└ @ Base loading.jl:894
┌ Warning: Replacing module `Colors`
└ @ Base loading.jl:894
```
The issue seem to be back
